celery_spider_cms.py

```diff
@@ -0,0 +1,194 @@
+# -*- coding: utf-8 -*-
+# Author: Christian Ariza <christian.ariza AT gmail [DOT] com>
+"""
+Celery version of the cms htcondor_es spider.
+This version has some major changes:
+    - Celery based
+    - The same function is used for either the queues and
+       history queries.
+    - The history checkpoint for the schedds is stored in Redis instead of a json file.
+    - Parallelism is managed by celery
+"""
+import os
+import logging
+import argparse
+import time
+from celery import group
+from htcondor_es.celery.tasks import query_schedd, create_affiliation_dir
+from htcondor_es.celery.celery import app
+from htcondor_es.utils import get_schedds, get_schedds_from_file
+
+logging.basicConfig(level=os.environ.get("LOGLEVEL", "DEBUG").upper())
+__TYPE_HISTORY = "history"
+__TYPE_QUEUE = "queue"
+
+
+def main_driver(args):
+    logging.debug(app.conf.humanize(with_defaults=False))
```

Comment:
put proper docstring for EVERY function. The doc strings can serve as documentation if we'll apply sphinx parser later.